search-improvements.rst

```diff
@@ -0,0 +1,189 @@
+search engine, as well as adding more features to our search functionality like exact match search,
+case insensitive search, search as you type, suggestions and more.
+
+Google Summer of Code
+^^^^^^^^^^^^^^^^^^^^^
+Google Summer of Code is a global program where students work with an open source organization
+on a 3 month programming project. The core team of Read The Docs proposed some `Project Ideas`_,
+one of them was **Refactor & improve our search code**. I (`Safwan Rahman`_) was keen to get my hand dirty with
+Elasticsearch_ and grasped the opportunity to do so by applying
+for this project and I got accepted.
+
+I have worked full time for last 3 months to upgrade the whole codebase
+to compatible with `Elasticsearch 6.x`_ and also implemented various features like:
+
+- `Exact Matching Search`_
+- `Case Insensitive Search`_
+- `Improved Result Order`_,
+- `Zero Downtime Indexing`_
+
+Together with this, I am willing to implement more features like following:
+
+- `Search as You Type and Autocomplete`_
+- `Code Search`_ 
+
+All of my search related work can be seen in the `Search Project Board`_.
+
+
+Background
+^^^^^^^^^^
+Search is a vital part of any documentation hosting platform, so people can get the
+information they need. As a documentation hosting platform, the same rules apply to
+Read The Docs. Because of having a small core team, the search functionality
+of Read The Docs has lagged behind for quite a while now. Initially the search code
+was voluntarily contributed by `Rob Hudson`_,  `back in 2013`_ and then improved by other
+contributors. The search infrastructure was already outdated as
+Read The Docs were using `Elasticsearch 1.3.x`_ which was already reached its
+End of Life in 2016. Therefore, the upgradation of search infrastructure was badly needed.
+
+Built in Search vs Read The Docs Search
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Both Sphinx and MkDocs already have built in search functionality. But the features are very limited.
+At Read The Docs, we have felt the limitations and therefore we index the documentations in our
+Elasticsearch_ index so that we can provide better search experience like:
+
+- Search across multiple projects
+- Advanced query syntax
+- Search inside subprojects
+- Improved search result order
+- Public Search API (Documentation pending)
+
+**Currently, we do not index MkDocs documents with** Elasticsearch_,
+but `any kind of help is welcome`_.
+
+New Features
+^^^^^^^^^^^^
+In the 3 months of full time work, I have implemented couple of features including
+many bug fixes. Some of the major features are as following:
+
+- `Exact Matching Search`_: Exact matching is one of our most highly requested
+  features for Read the Docs. Now you can search for exact word in documentations
+  by having your query inside quotation (`""`). So if you search
+  for `"Here is foo"` (with the quotation), you will get all the documentations where the full
+  `Here is foo` phrase exist.
```

Comment:
exist -> exists.

Single backticks don't really do anything on the stylesheet we are using. If you build the blog, you'll see this:

<img width="550" alt="screen shot 2018-10-04 at 1 25 00 pm" src="https://user-images.githubusercontent.com/185043/46500952-f0b42000-c7d8-11e8-8516-e604fd0de6f2.png">